### PR TITLE
Placeholder synth and waveform visualizer

### PR DESCRIPTION
**Pull request by frostburn**
ref #30

### CODE REVIEW

File: src/views/SynthView.vue [hunk 1]
```diff
@@ -31,14 +59,45 @@ function updateEquaveShift(event: Event) {
...
+          <input
+            type="range"
+            min="0"
+            max="0.5"
+            step="any"
+            v-model="mainVolume"
+          />
+        </div>
+        <h2>Waveform</h2>
+        <TimeDomainVisualizer
```
Comment (frostburn) [inline src/views/SynthView.vue:95]: This canvas element is ugly here. How do you even style those responsively? I'm pretty sure it has to be a canvas due to the amount of sound data it has to render.

Comment (frostburn) [reply src/views/SynthView.vue:95]: Used CSS width = 100% with offset width compensation while drawing the canvas.

**Comment by frostburn** [comment]
The order of the main lowpass and gain should be reversed to get visuals of the actual sound output.

**Comment by frostburn** [comment]
**Sevish** _— Today at 4:37 PM_

> It's worth adding a high pass as well just in case someone tries playing 5 mHz and blows out their speakers 😄
> a steep one at 30 Hz would prob be fine

**Comment by frostburn** [comment]
Added the highpass. Didn't set it too steep to avoid ringing. Could add another for 24db/octave if needed.

**Comment by SeanArchibald** [comment]
The default main volume is loud enough that if I hold 5 notes then I hear distortion. Maybe it can come down to 0.11 ish? At this value I get roughly equal loudness with SW1 triangle.

If I change the value of main volume slider, and then change the value of any isomorphic key mapping control, the main volume slider moves by itself (though the actual gain doesn't seem to get affected).

**Comment by frostburn** [comment]
> If I change the value of main volume slider, and then change the value of any isomorphic key mapping control, the main volume slider moves by itself (though the actual gain doesn't seem to get affected).

That seems like a bug in v-model I'll check if there's a workaround.

**Comment by frostburn** [comment]
Set default volume to 0.11 and fixed the volume slider moving by itself.

**Comment by frostburn** [comment]
Fixed the issues on Chromium.

**Review by SeanArchibald — APPROVED** [review]

**Pull request merged**